subsys/bluetooth/rpc/host/bt_rpc_gatt_host.c

```diff
@@ -950,3 +1065,617 @@ static void bt_gatt_cb_register_handler(CborValue *value, void *handler_data)
+{
+	data->uuid = bt_uuid_cli_dec(value, (struct bt_uuid *)data->uuid);
+	data->start_handle = ser_decode_uint(value);
+	data->end_handle = ser_decode_uint(value);
+	data->type = ser_decode_uint(value);
+}
+
+static void bt_gatt_discover_rpc_handler(CborValue *value, void *_handler_data)
+{
+	int result;
+	struct bt_conn *conn;
+	struct bt_gatt_discover_container* container;
+
+	container = k_malloc(sizeof(struct bt_gatt_discover_container));
+	if (container == NULL) {
+		ser_decoding_done_and_check(value);
+		goto alloc_error;
+	}
+	container->params.uuid = &container->uuid;
+
+	conn = bt_rpc_decode_bt_conn(value);
+	bt_gatt_discover_params_dec(value, &container->params);
+	container->remote_pointer = ser_decode_uint(value);
+
+	container->params.func = bt_gatt_discover_callback;
+
+	if (!ser_decoding_done_and_check(value)) {
+		goto decoding_error;
+	}
+
+	result = bt_gatt_discover(conn, &container->params);
+
+	if (result < 0) {
+		k_free(container);
+	}
+
+	ser_rsp_send_int(result);
+
+	return;
+
+decoding_error:
+	k_free(container);
+alloc_error:
+	report_decoding_error(BT_GATT_DISCOVER_RPC_CMD, _handler_data);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(bt_rpc_grp, bt_gatt_discover, BT_GATT_DISCOVER_RPC_CMD,
+	bt_gatt_discover_rpc_handler, NULL);
+
+uint8_t bt_gatt_read_callback(struct bt_conn *conn, uint8_t err,
+			      struct bt_gatt_read_params *params,
+			      const void *data, uint16_t length)
+{
+	uint8_t result;
+	struct bt_gatt_read_container* container;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	container = CONTAINER_OF(params, struct bt_gatt_read_container, params);
+
+	NRF_RPC_CBOR_ALLOC(ctx, 5 + 3 + 2 + 5 + 5 + length);
+
+	ser_encode_uint(&ctx.encoder, SCRATCHPAD_ALIGN(length));
+	bt_rpc_encode_bt_conn(&ctx.encoder, conn);
+	ser_encode_uint(&ctx.encoder, err);
+	ser_encode_uint(&ctx.encoder, container->remote_pointer);
+	ser_encode_buffer(&ctx.encoder, data, length);
+
+	nrf_rpc_cbor_cmd_no_err(&bt_rpc_grp, BT_GATT_READ_CALLBACK_RPC_CMD,
+		&ctx, ser_rsp_decode_u8, &result);
+
+	if (result == BT_GATT_ITER_STOP || data == NULL || params->handle_count == 1) {
+		k_free(container);
+	}
+
+	return result;
+}
+
+static void bt_gatt_read_params_dec(CborValue *value, struct bt_gatt_read_params *data)
+{
+	if (data->handle_count == 0) {
+		data->by_uuid.start_handle = ser_decode_uint(value);
+		data->by_uuid.end_handle = ser_decode_uint(value);
+		data->by_uuid.uuid = bt_uuid_cli_dec(value, (struct bt_uuid *)data->by_uuid.uuid);
+	} else if (data->handle_count == 1) {
+		data->single.handle = ser_decode_uint(value);
+		data->single.offset = ser_decode_uint(value);
+	} else {
+		ser_decode_buffer(value, data->multiple.handles,
+				  sizeof(data->multiple.handles[0]) * data->handle_count);
+		data->multiple.variable = ser_decode_bool(value);
+	}
+}
+
+static void bt_gatt_read_rpc_handler(CborValue *value, void *_handler_data)
+{
+	struct bt_conn * conn;
+	struct bt_gatt_read_container* container;
```

Comment:
```suggestion
	struct bt_gatt_read_container *container;
```

Reply:
Please check pointers formatting also in other places. The checkpath will complain about it